### PR TITLE
Enable short verification codes remotely and related UI facelift

### PR DESCRIPTION
**Pull request by i1skn**
### Description

Enable short verification codes remotely and related UI facelift

### Tested

iOS & Android both with enabled and disabled feature 

### Related issues

- Fixes https://github.com/celo-org/celo-monorepo/issues/6631

### Backwards compatibility

Yes

### Screenshots
Simulator Screen Shot - iPhone 11 - 2021-02-23 at 18 32 49
Simulator Screen Shot - iPhone 11 - 2021-02-23 at 18 32 43
Simulator Screen Shot - iPhone 11 - 2021-02-23 at 18 32 11

### CODE REVIEW

File: packages/mobile/src/components/CodeInput.tsx [hunk 1]
```diff
@@ -109,18 +136,24 @@ export default function CodeInput({
...
+                    textAlign: 'center',
+                  }),
+                  minHeight:
+                    Platform.OS === 'ios' && numberOfLines
+                      ? LINE_HEIGHT * numberOfLines
+                      : undefined,
+                }}
                 autoCapitalize="none"
                 testID={testID}
               />
```
Comment (jeanregisser) [inline packages/mobile/src/components/CodeInput.tsx:151]: I think we also want `keyboardType` to be `number-pad` when `shortVerificationCodesEnabled` is `true`, right?

**Review by jeanregisser — COMMENTED** [review]
Looking great! 👍 🚀 

A couple of points:
- can you test on Android too to be sure it doesn't break anything?
- could you add screenshots so it's easier to review the UI changes?

**Comment by i1skn** [comment]
@jeanregisser great suggestions! 
I've tested it on Android and also changed keyboard to numeric on iOS. 
Also have added screenshot to the description!

File: packages/mobile/src/components/CodeInput.tsx [hunk 2]
```diff
@@ -105,22 +132,34 @@ export default function CodeInput({
                 // on the native input. Though it doesn't work in all cases (see https://stackoverflow.com/a/33227237/158525)
                 // and has the unfortunate drawback of breaking multiline autosize.
                 // We use numberOfLines to workaround this last problem.
-                keyboardType={Platform.OS === 'android' ? 'visible-password' : undefined}
+                keyboardType={
+                  Platform.OS === 'android'
+                    ? 'visible-password'
+                    : shortVerificationCodesEnabled
+                    ? 'number-pad'
+                    : undefined
```
Comment (jeanregisser) [inline packages/mobile/src/components/CodeInput.tsx:140]: Don't we want `number-pad` too on Android?

Comment (i1skn) [reply packages/mobile/src/components/CodeInput.tsx:140]: I'm slightly worried about the comment above `autoCorrect={false}`, which starts with `// This disables keyboard suggestions on iOS, but unfortunately NOT on Android ...`. So, I'm hesitant to replace`visible-password` with `number-pad`. WDYT?

Comment (jeanregisser) [reply packages/mobile/src/components/CodeInput.tsx:140]: Yes I was fighting to disable `autoCorrect` on Android.
But here using `number-pad` should be safe on both iOS and Android when `shortVerificationCodesEnabled` is `true`.
And we can keep the existing logic when it's `false`.

**Review by jeanregisser — APPROVED** [review]

**Comment by Celoqa** [comment]
Hi @i1skn we have verified this task using latest Android Internal Build V1.12.0(1004294340) & Test flight build V1.12.0(49) and observed the followings.
- User is able to receive 8 digit verification codes and able to compete the verification flow.

**User is able to receive 8 digit verification codes and able to complete the verification flow when user go through below steps:**
- Able to receive 8 digits verification codes & able to complete verification flow on-Boarding flow.
- Able to complete verification flow when user go through restore my account key.
- Able to complete verification flow when user go through Confirm phone number in-App notification.
- Able to complete verification flow when user go through settings> Confirm phone number 
- Able to complete verification flow when user go through Earn rewards> Confirm phone number.

Please let us know if we need to test anything else in this task.

**Comment by i1skn** [comment]
@Celoqa Thanks for checking it, can you please open an issue with attached logs/video, so I can check is that an expected behaviour or not. Thanks!

**Comment by Celoqa** [comment]
Sure @i1skn We will raise a new issue with attached logs.

**Comment by Celoqa** [comment]
Hi @i1skn We have raised a new issue with attached logs #146

**Comment by i1skn** [comment]
Thanks!

**Comment by Celoqa** [comment]
Hi @i1skn Can you please confirm below issue is expected Or not if not we can raise a new issue.

- Confirmation failed message is shown when user change the phone number after receiving attestation.

**Build:** Android Internal Build V1.12.0(1004294340) & Test flight build V1.12.0(49)
**Devices:** iPhone 6+ (12.4.5), Samsung M30(10)
**Repro steps:**
- Install the app
- Go through on-boarding flow
- Enter phone number
- Tap on start button & complete captcha
- Make sure user should receive attestation 
- Go back to Phone number page 
- Change the phone number
- Tap on resume button
- Observe 
**Attachment:**
- [Confirmation Failed.mp4](https://drive.google.com/file/d/1Mvd8o2jnTEF-IfxwXTcnfWwjnf7I6PX9/view?usp=sharing)
- [ConfirmationFailedLogs](https://drive.google.com/file/d/19IvZGAfJXrkaN15_1dHiyHZ22vv_pkOu/view?usp=sharing)

**Pull request merged**